topy/topy.py

```diff
@@ -121,11 +121,27 @@ def sanitize_filename(filename):
 def print_diff(filename, old, new, stream=sys.stdout):
     """Diffs the `old` and `new` strings and prints as unified diff to file-like object `stream`."""
 
-    # TODO: color output for terminals
     lines = unified_diff(old.splitlines(True), new.splitlines(True), filename, filename)
+    lines = map(lambda l: determine_color(l, stream=stream), lines)
```

Comment:
Currently `determine_color()` calls `is_color_output_required()` for every line, but you only need to check it once:
```py
if is_color_output_required(stream):
    lines = map(add_output_color, lines)
```

Reply:
Good spot, I'll update this